python-backend/app/api/mines/reports/models/mine_report.py

```diff
@@ -74,3 +73,15 @@ def find_by_mine_report_guid(cls, _id):
             return cls.query.filter_by(mine_report_guid=_id).first()
         except ValueError:
             return None
+
+    @classmethod
+    def find_by_mine_guid_with_category(cls, _id, category):
+        try:
+            uuid.UUID(_id, version=4)
+            reports = cls.query.filter_by(mine_guid=_id).all()
+            return [
+                r for r in reports if category.upper() in
+                [c.mine_report_category.upper() for c in r.mine_report_definition.categories]
```

Comment:
If i read this correctly, is this:
Return a list of reports whose category belongs to one or more of the categories this mine report belongs in in the list of mine report definition categories?

I may be wrong but the inner list may get formed and looped over for every single report and you can probably avoid that by defining the inner list outside of the return statement and save some iterative cycles.

Reply:
I think that's true, I took the first pass and got something that was easier to follow. Can be improved in the future.

Reply:
No, the `for c in r.mine_report_definition.categories` is generating the list of categories of the selected report, not the list of categoreis to filter by. 

This is a pain to read, but I think its right. Its doing this:
```
for report in reports
   for report_category in report.categories
       return report_category == filter_category
```